libraries/AP_Motors/examples/AP_Motors_test/AP_Motors_test.cpp

```diff
@@ -30,105 +27,175 @@ void stability_test();
+    AP_Motors::motor_frame_class frame_class = AP_Motors::MOTOR_FRAME_QUAD;
 
-#if HELI_TEST == 0
-    motors.update_throttle_range();
-    motors.set_throttle_avg_max(0.5f);
-#endif
-    motors.output_min();
-
-    // allow command line args for single run
+    // Parse the command line arguments
     uint8_t argc;
     char * const *argv;
     hal.util->commandline_arguments(argc, argv);
     if (argc > 1) {
-#if HELI_TEST == 0
         for (uint8_t i = 2; i < argc; i++) {
-            const char* arg = argv[i];
-
+            const char *arg = argv[i];
             const char *eq = strchr(arg, '=');
+
             if (eq == NULL) {
                 ::printf("Expected argument with \"=\"\n");
                 exit(1);
             }
+
             char cmd[20] {};
             strncpy(cmd, arg, eq-arg);
             const float value = atof(eq+1);
             if (strcmp(cmd,"yaw_headroom") == 0) {
-                motors.set_yaw_headroom(value);
+                if (motors_matrix != nullptr) {
+                    motors_matrix->set_yaw_headroom(value);
+                } else {
+                    ::printf("frame_class = %d does not accept yaw_headroom commands\n", frame_class);
+                }
 
             } else if (strcmp(cmd,"throttle_avg_max") == 0) {
-                motors.set_throttle_avg_max(value);
+                if (motors_matrix != nullptr) {
+                    motors_matrix->set_throttle_avg_max(value);
+                } else {
+                    ::printf("frame_class = %d does not accept throttle_avg_max commands\n", frame_class);
+                }
 
             } else if (strcmp(cmd,"thrust_boost") == 0) {
                 thrust_boost = value > 0.0;
 
+            } else if (strcmp(cmd,"frame_class") == 0) {
+                // We must have the frame_class argument 2nd as resulting class is used to determine if
+                // we have access to certain functions in the multicopter motors child class
+                if (i != 2) {
+                    ::printf("frame_class must be second argument\n");
+                    exit(1);
+                }
+
+                // Setup the correct motors object for the frame class to test
+                frame_class = (AP_Motors::motor_frame_class)value;
+
+                switch (frame_class) {
+
+                    case AP_Motors::MOTOR_FRAME_QUAD:
+                    case AP_Motors::MOTOR_FRAME_HEXA:
+                    case AP_Motors::MOTOR_FRAME_OCTA:
+                        motors_matrix = new AP_MotorsMatrix(400);
+                        motors = motors_matrix;
+                        num_outputs = __builtin_popcount(motors->get_motor_mask());
+                        break;
+
+                    case AP_Motors::MOTOR_FRAME_HELI:
+                        motors = new AP_MotorsHeli_Single(400);
+                        num_outputs = 6; // Mot 1-3 swashplate, mot 4 tail rotor pitch, mot 6 for 4th servo in H4-90 swash
```

Comment:
I think there are only 5 possible servos for heli_single.  Mot 1-3 swashplate with mot 5 added for 4 servo swashplates.  mot 4 is tail rotor.  Not sure how the ext_gyro is handles.  Looks like it adds a motor.